cleanSweStore.pl

```diff
@@ -0,0 +1,169 @@
+
+=head1 SYNOPSIS
+
+ cleanSweStore.pl -help|-man
+ cleanSweStore.pl -projects <file with list of projects> [-execute]
+
+=head1 OPTIONS
+
+=over 4
+
+=item -h|-help
+
+prints out a brief help text.
+
+=item -m|-man
+
+Opens the manpage.
+
+=item -runfolder
+
+Path to a file containing all projects that should be removed from SweStore.
+
+Format, runfolder and projectID seperated by tab: runfoldername1	projectId1
+
+=item -execute
+
+Use flag when you want to perform the deletion. If not set, the script
+will only validate that the provided projects can be found and also
+specify if the entire runfolder will be deleted or just a subset
+of the projects.
+
+=back
+
+=cut
+
+my ($project, $debug, $execute);
+
+my $swestorePath = "/ssUppnexZone/proj/a2009002";
+
+my ($help,$man) = (0,0);
+
+# Project input file should have the following format
+# runfoldername1	projectId1
+# runfoldername1	projectId2
+# runfoldername2	projectId3
+# ...
+# runfoldernameN\tprojectIdN
+#
+
+GetOptions('help|?'=>\$help,
+	   'man'=>\$man,
+	   'projects=s' => \$project, 
+	   'execute!' => \$execute,
+    	   'debug' => \$debug,
+      	    ) or pod2usage(-verbose => 0);
+
+pod2usage(-verbose => 1)  if ($help);
+pod2usage(-verbose => 2)  if ($man);
+unless (defined($project)) {
+	print "You must provide a list of projects to clean, format: runfoldername\tprojectid\n";
+	pod2usage(-verbose => 1);
+	exit;
+}
+
+#
+# Save all runfolder and the associated projects into a hash structure
+#
+# var => {
+#	runfolder_name1 => {
+#		projId1 => 1,
+#		projId2 => 1,
+#	}
+#	runfolder_name2 => {
+#		projId3 => 1,
+#	        projId4 => 1,
+#	}#
+# }
+#
+open PROJECTS, $project or die "Couldn't open project file: $project!";
+my $dataToClean;
+while(<PROJECTS>){
+	if(!/^#/) {
+		chomp;
+		my ($runfolder,$project) = split(/\t/, $_);
+		$project =~ s/\s+//;
+		$dataToClean->{$runfolder}->{$project} = 1;
+	}
+}
+close(PROJECTS);
+
+my $timestamp = time;
+
+open REMOVED, "> removedFromSweStore.$timestamp.log" or die "Couldn't open output file: removedFromSweStore.$timestamp.log!\n";
+open SAVED, "> leftOnSweStore.$timestamp.log" or die "Couldn't open output file: leftOnSweStore.$timestamp.log!\n";
+
+print "Cleaning swestore!\n";
+#
+# Remove the provided projects from SweStore
+#
+foreach my $runfolder (keys %{$dataToClean}) { # Process each runfolder 
+	my ($year,$month,$day) = ($runfolder =~ m/^(\d{2})(\d{2})(\d{2})_[A-Z0-9]+_[0-9]+_[A-Z0-9]/);
+	# Find each  stored project at SweStore, for the specified runfolder
+	my $projects = qx(ils $swestorePath/20$year-$month/$runfolder/Projects/);
+	#Result from ils
+	#
+	# ils /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/
+	# /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects:
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0401
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0429
+	#
```

Comment:
if doing extensive irods magic from perl, it may be worth looking into some [api wrapper](https://github.com/wtsi-npg/perl-irods-wrap). not saying it's necessary for this task though